src/io.jl

```diff
@@ -99,8 +99,14 @@ function Base.write(io::IO, d::MRCData; compress = :none)
     T = datatype(h)
     data = parent(d)
     fswap = bswapfromh(h.machst)
-    for i in eachindex(data)
-        @inbounds sz += write(newio, fswap(T(data[i])))
+    unit_vsize = 4096 ÷ sizeof(T)
+    vlen = length(data)
+    vrem = vlen % unit_vsize
+    if vrem != 0
+        @inbounds sz += write(newio, fswap.(T.(data[1:vrem])))
```

Comment:
This should avoid an allocation
```suggestion
        @inbounds @views sz += write(newio, fswap.(T.(data[1:vrem])))
```